portfolio/src/main/java/com/google/sps/servlets/favBirdDataServlet.java

```diff
@@ -29,21 +35,42 @@
 @WebServlet("/favorite-bird")
 public class favBirdDataServlet extends HttpServlet {
 
-  private Map<String, Integer> birdVotes = new HashMap<>();
-
+  /**
+  * Fetches data with a query, sorts data into a hash map, 
+  * and translates to JSON to load to the graph
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    //create and prepare a query
+    Query query = new Query("Bird");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
```

Comment:
so, the way this is run, every time we want to see the number of bird votes, we need to query the entire database. this would get expensive if we have a lot of votes in the database (or a lot of birds, if you make the changes suggested below). Instead, what if we could query the entire database once upon the server starting up, store that variable locally, then keep updating the database AND the local variable as new votes are added. That way, when users ask for the current votes, we can return them the local variable. But if the server goes down, we have everything stored in the database so when we load the server back up, we can grab the latest data again, store it locally, and act like nothing happened:

https://stackoverflow.com/a/18808918

If this is confusing, let me know, would love to chat about this over GVC.